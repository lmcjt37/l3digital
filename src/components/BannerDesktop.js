import React from 'react'
import PropTypes from 'prop-types'

import Banner from 'components/Banner'

class BannerDesktop extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            beginAnimation: '',
        }
    }

    componentDidMount() {
        this.setState({ beginAnimation: 'go' })
    }

    componentWillUnmount() {
        this.setState({ beginAnimation: '' })
    }

    render() {
        return (
            <Banner classes="major front-page">
                <div className="inner">
                    <header className="major">
                        <svg
                            className={'intro ' + this.state.beginAnimation}
                            viewBox="0 0 300 60"
                            xmlns="http://www.w3.org/2000/svg"
                            version="1.1"
                        >
                            <clipPath id="rect1">
                                <rect x="0" y="26" width="100%" height="1" />
                            </clipPath>
                            <clipPath id="text1">
                                <text
                                    textAnchor="start"
                                    x="0"
                                    y="15"
                                    className="text"
                                >
                                    We Are L3 Digital
                                </text>
                            </clipPath>

                            <rect
                                x="0"
                                y="26"
                                width="100%"
                                height="1"
                                className="base-stroke"
                                clipPath="url(#rect1)"
                            />
                            <rect
                                x="0"
                                y="26"
                                width="100%"
                                height="1"
                                className="base-stroke rect-stroke"
                                clipPath="url(#rect1)"
                            />
                            <text
                                textAnchor="start"
                                x="0"
                                y="15"
                                className="text base-stroke"
                                clipPath="url(#text1)"
                            >
                                We Are L3 Digital
                            </text>
                            <text
                                textAnchor="start"
                                x="0"
                                y="15"
                                className="text base-stroke text-stroke"
                                clipPath="url(#text1)"
                            >
                                We Are L3 Digital
                            </text>
                        </svg>
                    </header>
                    <div className="content">
                        <p>
                            A team of great collaborators, providing solutions
                            to all your development needs.
                        </p>
                        <ul className="actions">
                            <li>
                                <a
                                    href="javascript:;"
                                    className="button next"
                                    onClick={this.props.scrollToElement}
                                >
                                    Learn more
                                </a>
                            </li>
                        </ul>
                    </div>
                </div>
                <div className="scroll">
                    <a
                        href="javascript:;"
                        onClick={this.props.scrollByScreenHeight}
                    >
                        <span />
                        Scroll
                    </a>
                </div>
            </Banner>
        )
    }
}

BannerDesktop.propTypes = {
    scrollToElement: PropTypes.func,
    scrollByScreenHeight: PropTypes.func,
}

export default BannerDesktop
