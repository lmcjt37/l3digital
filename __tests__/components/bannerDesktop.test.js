import BannerDesktop from 'components/BannerDesktop'

jest.mock('helpers/utils')

describe('BannerDesktop tests', () => {
    test('check it renders correctly with snapshot', () => {
        const tree = render(<BannerDesktop />)
        expect(tree).toMatchSnapshot()
    })
})
