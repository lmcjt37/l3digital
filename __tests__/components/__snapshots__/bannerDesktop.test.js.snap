// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerDesktop tests check it renders correctly with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="major front-page"
      >
        <div
          class="inner"
        >
          <header
            class="major"
          >
            <svg
              class="intro go"
              version="1.1"
              viewBox="0 0 300 60"
              xmlns="http://www.w3.org/2000/svg"
            >
              <clippath
                id="rect1"
              >
                <rect
                  height="1"
                  width="100%"
                  x="0"
                  y="26"
                />
              </clippath>
              <clippath
                id="text1"
              >
                <text
                  class="text"
                  text-anchor="start"
                  x="0"
                  y="15"
                >
                  We Are L3 Digital
                </text>
              </clippath>
              <rect
                class="base-stroke"
                clip-path="url(#rect1)"
                height="1"
                width="100%"
                x="0"
                y="26"
              />
              <rect
                class="base-stroke rect-stroke"
                clip-path="url(#rect1)"
                height="1"
                width="100%"
                x="0"
                y="26"
              />
              <text
                class="text base-stroke"
                clip-path="url(#text1)"
                text-anchor="start"
                x="0"
                y="15"
              >
                We Are L3 Digital
              </text>
              <text
                class="text base-stroke text-stroke"
                clip-path="url(#text1)"
                text-anchor="start"
                x="0"
                y="15"
              >
                We Are L3 Digital
              </text>
            </svg>
          </header>
          <div
            class="content"
          >
            <p>
              A team of great collaborators, providing solutions to all your development needs.
            </p>
            <ul
              class="actions"
            >
              <li>
                <a
                  class="button next-effect effect"
                  href="javascript:;"
                >
                  Learn more
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="scroll"
        >
          <a
            href="javascript:;"
          >
            <span />
            Scroll
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="major front-page"
    >
      <div
        class="inner"
      >
        <header
          class="major"
        >
          <svg
            class="intro go"
            version="1.1"
            viewBox="0 0 300 60"
            xmlns="http://www.w3.org/2000/svg"
          >
            <clippath
              id="rect1"
            >
              <rect
                height="1"
                width="100%"
                x="0"
                y="26"
              />
            </clippath>
            <clippath
              id="text1"
            >
              <text
                class="text"
                text-anchor="start"
                x="0"
                y="15"
              >
                We Are L3 Digital
              </text>
            </clippath>
            <rect
              class="base-stroke"
              clip-path="url(#rect1)"
              height="1"
              width="100%"
              x="0"
              y="26"
            />
            <rect
              class="base-stroke rect-stroke"
              clip-path="url(#rect1)"
              height="1"
              width="100%"
              x="0"
              y="26"
            />
            <text
              class="text base-stroke"
              clip-path="url(#text1)"
              text-anchor="start"
              x="0"
              y="15"
            >
              We Are L3 Digital
            </text>
            <text
              class="text base-stroke text-stroke"
              clip-path="url(#text1)"
              text-anchor="start"
              x="0"
              y="15"
            >
              We Are L3 Digital
            </text>
          </svg>
        </header>
        <div
          class="content"
        >
          <p>
            A team of great collaborators, providing solutions to all your development needs.
          </p>
          <ul
            class="actions"
          >
            <li>
              <a
                class="button next-effect effect"
                href="javascript:;"
              >
                Learn more
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="scroll"
      >
        <a
          href="javascript:;"
        >
          <span />
          Scroll
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
